frontends/concrete-python/concrete/fhe/compilation/compiler.py

```diff
@@ -537,5 +537,9 @@ def compile(
         finally:
             self.configuration = old_configuration
             self.artifacts = old_artifacts
+            if self.graph and len(self.graph.output_nodes) > 1:
```

Comment:
And why is this in the finally block and not after it?

Reply:
because of the return I guess
But I can move the return and to if in the else block

Reply:
done